### PR TITLE
Upgrade ICL eval to use `ComposerHFCausalLM`

### PR DESCRIPTION
**Pull request by abhi-mosaic**
This PR:
* removes `model_loading.py`
* standardized `cfg.model` specification for eval and training configs
* fixes `ComposerHFCausalLM` to work with ICL metrics

### CODE REVIEW

**Review by dakinggg — APPROVED** [review]
I think this looks right to me. but obviously real test is just running all the things. if you've done that, feel free to merge.

File: examples/llm/src/hf_causal_lm.py [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
 
 from examples.common.hf_fsdp import prepare_hf_causal_lm_model_for_fsdp
 
 
 class ComposerHFCausalLM(HuggingFaceModel):
 
     def __init__(self, cfg: DictConfig):
-        config = AutoConfig.from_pretrained(cfg.hf_config_name_or_path)
-        tokenizer = AutoTokenizer.from_pretrained(cfg.hf_config_name_or_path)
+        config = AutoConfig.from_pretrained(cfg.pretrained_model_name_or_path)
```
Comment (alextrott16) [inline examples/llm/src/hf_causal_lm.py:19]: This class should really have a docstring that explains what the expected config fields are and what they mean.

File: examples/llm/src/hf_causal_lm.py [hunk 2]
```diff
@@ -42,20 +41,24 @@ def get_targets(self, batch: dict):
         targets[:, -1] = -100
         return targets
 
-    def forward(self, batch: dict):
+    def forward(self, batch: dict) -> CausalLMOutput:
         return self.model(input_ids=batch['input_ids'],
-                          attention_mask=batch['attention_mask'].bool()).logits
-
-    def eval_forward(self, batch: dict, outputs: Optional[Tensor] = None):
-        return outputs if outputs is not None else self.forward(batch)
```
Comment (alextrott16) [inline examples/llm/src/hf_causal_lm.py:50]: Be careful about switching to the parent `eval_forward` without changing some of the other methods in this class. I'm specifically worried about how the parent `eval_forward` pops `labels` out of the batch and stashes it in `self.labels`. This might require some re-writing of this class's `get_targets` and maybe `loss` too.

Comment (abhi-mosaic) [reply examples/llm/src/hf_causal_lm.py]: Reverting this...

File: examples/llm/src/hf_causal_lm.py (hunk 1 above)
Comment (abhi-mosaic) [reply examples/llm/src/hf_causal_lm.py:22]: Punting to a future "Add Docstrings" PR

**Comment by abhi-mosaic** [comment]
All manual checks passing!
